cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,161 @@
+    def setUp(self):
+        self.client = APIClient()
+        self.user = User.objects.create_user(
+            email="testuser@example.com", password="testpass"
+        )
+        self.admin_user = User.objects.create_superuser(
+            email="adminuser@example.com", password="adminpass"
+        )
+        self.token = Token.objects.create(user=self.user)
+        self.admin_token = Token.objects.create(user=self.admin_user)
+        self.client.credentials(HTTP_AUTHORIZATION=f"Token {self.token.key}")
+
+        self.genre1 = create_genre("Action")
+        self.genre2 = create_genre("Drama")
+
+        self.actor1 = create_actor("John", "Doe")
+        self.actor2 = create_actor("Jane", "Smith")
+
+        self.movie1 = create_movie(
+            "Movie One", genres=[self.genre1], actors=[self.actor1]
+        )
+        self.movie2 = create_movie(
+            "Movie Two", genres=[self.genre2], actors=[self.actor2]
+        )
+
+    def test_list_movies(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+
+    def test_filter_movies_by_title(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"title": "Movie One"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_genres(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"genres": f"{self.genre1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_actors(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"actors": f"{self.actor1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_create_movie_as_authenticated_user(self):
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        expected_error = {
+            "detail": ErrorDetail(
+            string="You do not have permission to perform this action.",
+            code="permission_denied"
+            )
+        }
+        self.assertEqual(response.data, expected_error)
```

Comment:
You have a test case for creating a movie as an authenticated user, but what about an unauthenticated user? It would be good to add a test case to ensure that unauthenticated users cannot create movies.